src/gt4py/backend/python_generator.py

```diff
@@ -169,21 +169,21 @@ def visit_VarRef(self, node: gt_ir.VarRef):
 
         return source
 
-    def visit_UnaryOpExpr(self, node: gt_ir.UnaryOpExpr):
+    def visit_UnaryOpExpr(self, node: gt_ir.UnaryOpExpr, **kwargs):
         fmt = "({})" if isinstance(node.arg, gt_ir.CompositeExpr) else "{}"
         source = "{op} {expr}".format(
-            op=self.OP_TO_PYTHON[node.op], expr=fmt.format(self.visit(node.arg))
+            op=self.OP_TO_PYTHON[node.op], expr=fmt.format(self.visit(node.arg, **kwargs))
         )
 
         return source
 
-    def visit_BinOpExpr(self, node: gt_ir.BinOpExpr):
+    def visit_BinOpExpr(self, node: gt_ir.BinOpExpr, **kwargs):
         lhs_fmt = "({})" if isinstance(node.lhs, gt_ir.CompositeExpr) else "{}"
         rhs_fmt = "({})" if isinstance(node.rhs, gt_ir.CompositeExpr) else "{}"
         source = "{lhs} {op} {rhs}".format(
-            lhs=lhs_fmt.format(self.visit(node.lhs)),
+            lhs=lhs_fmt.format(self.visit(node.lhs, **kwargs)),
             op=self.OP_TO_PYTHON[node.op],
-            rhs=rhs_fmt.format(self.visit(node.rhs)),
+            rhs=rhs_fmt.format(self.visit(node.rhs, **kwargs)),
         )
 
         return source
```

Comment:
Recommend adding `**kwargs` to `visit_NativeFuncCall`...

Reply:
Good catch!